instrumentation/opentelemetry-instrumentation-pika/src/opentelemetry/instrumentation/pika/pika_instrumentor.py

```diff
@@ -126,12 +132,27 @@ def uninstrument_channel(channel: BlockingChannel) -> None:
             return
 
         for consumers_tag, client_info in channel._consumer_infos.items():
-            if hasattr(client_info.on_message_callback, "_original_callback"):
+            callback_attr = (
+                PikaInstrumentor._consumer_callback_attribute_name()
+            )
+            consumer_callback = getattr(client_info, callback_attr)
+            if hasattr(consumer_callback, "_original_callback"):
                 channel._consumer_infos[
                     consumers_tag
-                ] = client_info.on_message_callback._original_callback
+                ] = consumer_callback._original_callback
         PikaInstrumentor._uninstrument_channel_functions(channel)
 
+    @staticmethod
+    def _consumer_callback_attribute_name() -> str:
+        pika_version = version.parse(
+            pkg_resources.get_distribution("pika").version
+        )
+        return (
+            "on_message_callback"
+            if pika_version >= version.parse("1.0.0")
+            else "consumer_cb"
+        )
```

Comment:
I think that if the pika package ever has a breaking change in the future (like they apparently had between `1.0.0` and `1.1.0`) and change `on_message_callback` to `on_message_cb` or whatever, this will break, and since we don't have an actual test to detect this, this will go undetected.
This function should check `hasattr` on the object, rather than checking the version of imported pika.

Reply:
If a newer version is released with a breaking change, unit tests will fail - `getattr` throws `AttributeError` same as the dot operator. Also, I'm not changing the current behavior.